### PR TITLE
fixes #33439. The feature now changes according to the feature profile.

### LINKED ISSUE #33439: Still able to create 3D Scenes after disabling 3D in "Manage Editor Features"
**Godot version:**
3.2.beta 31f327a4a9c2637a8d2ada01b293ed138c3e6f99
**OS/device including version:**
5.3.8-arch1-1
**Issue description:**
After disabling "3D Editor" and all "Spatial" nodes in Manage Editor Features the Scene panel still gives the option to create a 3D Scene. By clicking on this option the scene's root node becomes a Spatial node. 

Since all Spatial nodes and the 3D editor are disabled the option to create a 3D Scene should be disabled as well
**Steps to reproduce:**
1. In the editor navigate to Editor > Manage Editor Features
2. Click New to create a new profile
3. Uncheck Enabled Features > 3D Editor and Enabled Classes > Nodes > Spatial
4. Click Make Current to enable the new profile
2019-11-08-042054_1920x1080_scrot
5. In the Scene panel click "3D Scene" to create a scene with a Spatial root node
2019-11-08-042352_1920x1080_scrot

### PR DESCRIPTION
**Pull request by code-xD**
Need to make the dock tree more synchronised with the feature profile manager.

*Bugsquad edit: This closes #33439.*

### CODE REVIEW

File: editor/scene_tree_dock.cpp [hunk 1]
```diff
@@ -1038,7 +1038,9 @@ void SceneTreeDock::_node_collapsed(Object *p_obj) {
 void SceneTreeDock::_notification(int p_what) {
 
 	switch (p_what) {
-
+		char* s;	
+		sprintf(s,"p_what - %d \n",p_what);
+		print_line(s);
```
Comment (YeldhamDev) [inline editor/scene_tree_dock.cpp]: I assume this is a leftover of debugging, please remove it.

File: editor/scene_tree_dock.cpp [hunk 2]
```diff
@@ -1099,14 +1101,14 @@ void SceneTreeDock::_notification(int p_what) {
 			button_2d->set_text(TTR("2D Scene"));
 			button_2d->set_icon(get_icon("Node2D", "EditorIcons"));
 			button_2d->connect("pressed", this, "_tool_selected", make_binds(TOOL_CREATE_2D_SCENE, false));
-
-			Button *button_3d = memnew(Button);
+
```
Comment (YeldhamDev) [inline editor/scene_tree_dock.cpp]: Also remove the trailing space.

File: editor/scene_tree_dock.cpp [hunk 3]
```diff
@@ -1099,14 +1101,14 @@ void SceneTreeDock::_notification(int p_what) {
...
-			Button *button_3d = memnew(Button);
+			
+			button_3d = memnew(Button);
 			beginner_node_shortcuts->add_child(button_3d);
 			button_3d->set_text(TTR("3D Scene"));
 			button_3d->set_icon(get_icon("Spatial", "EditorIcons"));
 			button_3d->connect("pressed", this, "_tool_selected", make_binds(TOOL_CREATE_3D_SCENE, false));
 
-			Button *button_ui = memnew(Button);
+			Button *button_ui = memnew(Button);
```
Comment (YeldhamDev) [inline editor/scene_tree_dock.cpp]: Also here.

File: editor/scene_tree_dock.cpp [hunk 4]
```diff
@@ -2752,12 +2754,15 @@ void SceneTreeDock::_feature_profile_changed() {
 
 		profile_allow_editing = !profile->is_feature_disabled(EditorFeatureProfile::FEATURE_SCENE_TREE);
 		profile_allow_script_editing = !profile->is_feature_disabled(EditorFeatureProfile::FEATURE_SCRIPT);
-
+		bool profile_allow_3d = !profile->is_feature_disabled(EditorFeatureProfile::FEATURE_3D);
+		
+		button_3d->set_visible(profile_allow_3d);
 		button_add->set_visible(profile_allow_editing);
-		button_instance->set_visible(profile_allow_editing);
+		button_instance->set_visible(profile_allow_editing);
```
Comment (YeldhamDev) [inline editor/scene_tree_dock.cpp]: And here.

**Comment by YeldhamDev** [comment]
If you desire to keep helping the engine, consider adding this hooks to your `.git/` folder, just follow the instructions: https://github.com/godotengine/godot/tree/master/misc/hooks

You may also want to check the code guidelines: https://docs.godotengine.org/en/3.1/community/contributing/code_style_guidelines.html

**Comment by YeldhamDev** [comment]
One last trailing space left.

Also, when you're done, squash your commits: https://docs.godotengine.org/en/3.1/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase

**Comment by code-xD** [comment]
@YeldhamDev is it why my travis tests are failing?

**Comment by YeldhamDev** [comment]
Yes, it also checks for formatting.

File: editor/scene_tree_dock.h [hunk 5]
```diff
@@ -112,6 +112,8 @@ class SceneTreeDock : public VBoxContainer {
 	ToolButton *button_create_script;
 	ToolButton *button_clear_script;
 
+	Button* button_3d;
```
Comment (YeldhamDev) [inline editor/scene_tree_dock.h]: Also, this `*` should be with `button_3d`, not `Button`.

Comment (code-xD) [reply editor/scene_tree_dock.h]: I am sorry for the inconveniences caused. I am new to open source development have usually worked on small scale projects.

Comment (YeldhamDev) [reply editor/scene_tree_dock.h]: Don't worry, we all have been there. You'll get used to it in no time!

**Comment by YeldhamDev** [comment]
You accidentally mixed some older commits into your branch.

**Comment by code-xD** [comment]
@YeldhamDev you can review the PR now. I have fixed the formatting errors. I guess it fulfils the required guidelines.

**Comment by code-xD** [comment]
> You accidentally mixed some older commits into your branch.

How should I resolve it? Should I fixup some more commits?

**Comment by YeldhamDev** [comment]
I think it's better to close this push and try again.

**Pull request closed without merging**